pkg/component/kubernetes/apiserver/apiserver.go

```diff
@@ -285,7 +285,8 @@ func (k *kubeAPIServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if err := k.reconcileHVPA(ctx, hvpa, deployment); err != nil {
+	// TODO(plkokanov): remove this after gardener v1.109.0 has been released.
+	if err := kubernetesutils.DeleteObject(ctx, k.client.Client(), hvpa); err != nil {
```

Comment:
When I think about it again, we should be able to clean up everything or said with other words assume that HVPA CRDs and CRs are now removed:
- In v1.105.0 with https://github.com/gardener/gardener/pull/10599 the VPAForETCD and VPAAndHPAForAPIServer feature gates are GA and locked to true. This means that when/if all Shoots/Garden resources successfully reconciled with v1.105.0, there will be no HVPA CRs in the landscape.
- In v1.106.0 with https://github.com/gardener/gardener/pull/10659 we deprecated the HVPA feature gates and locked them to false. This means that when/if all Seeds/Garden resources successfully reconciled with v1.106.0 there will be no HVPA CRDs in the landscape.

The corresponding handling for destroying the HVPA CRDs when HVPA feature is disabled:
- https://github.com/gardener/gardener/blob/ee82b03bf472b97a9316cea888aba4af3a6e9f8e/pkg/gardenlet/controller/seed/seed/components.go#L138-L141
- https://github.com/gardener/gardener/blob/6166ef7c551f8a2f793eb552a433bbfeeec82375/pkg/operator/controller/garden/garden/components.go#L161-L164

Preserving such a cleanup logic would help only if someone upgrades from v1.104 to v1.108 (the version this PR targets) and only if in v1.104 the HVPA features gates were enabled and the `VPAForETCD`/`VPAAndHPAForAPIServer` were explicitly disabled.

According to the docs skipping minor versions when upgrading Gardener is not recommended and we don't test these upgrade paths: https://github.com/gardener/gardener/blob/bbbc6b75e5ba75854f19c6a0dbba2f37d103aa61/docs/deployment/version_skew_policy.md#L19.

Let me know if I misssing something but I would suggest cleaning up the handlings instead of adding deletion logic that later on has to be cleaned up. Let me know if I miss something!

Reply:
For referenced, I pushed https://github.com/gardener/gardener/pull/10796/commits/e2ca9eeb5f287822d867a679716c6f2677335e90.

Reply:
I also added a release note of type `breaking operator` to described that folks first have to upgrade to v1.106.0 before upgrading to the version that includes this PR.